docs/dev_guides/new_library_comparisons_and_levels.md

```diff
@@ -0,0 +1,286 @@
+                length_difference=length_thres,
+                m_probability=m_prob,
+            )
+            comparison_levels.append(level)
+
+        comparison_levels.append(
+            self._else_level(m_probability=m_probability_else),
+        )
+
+        comparison_desc = ""
+        if include_exact_match_level:
+            comparison_desc += "Exact match vs. "
+
+        thres_desc = ", ".join(thresholds)
+        plural = "" if len(thresholds) == 1 else "s"
+        comparison_desc += (
+            f"Array length differences with threshold{plural} {thres_desc} vs. "
+        )
+        comparison_desc += "anything else"
+
+        comparison_dict = {
+            "comparison_description": comparison_desc,
+            "comparison_levels": comparison_levels,
+        }
+        super().__init__(comparison_dict)
+```
+
+Crucially we needed to use `self._null_level`, `self._exact_match_level`, `self._else_level` which already exist,
+but also the new `self._array_length_level` which relates to our new comparison level.
+We will need to make sure that the dialect-specific comparisons which will actually be _used_ will have this property.
+
+Each dialect has a comparison properties base, which stores information about all of the dialect-specific comparison levels used by all comparisons.
+We will need to add our new level to this, which we referred to above in `ArrayLengthAtThresholdsComparisonBase` - for this example in [`splink.spark.spark_comparison_library.py`](https://github.com/moj-analytical-services/splink/blob/master/splink/spark/spark_comparison_library.py):
+```python
+from splink.spark.spark_comparison_level_library import (
+    exact_match_level,
+    ...
+    array_length_level,
+)
+...
+
+class SparkComparisonProperties(SparkBase):
+    @property
+    def _exact_match_level(self):
+        return exact_match_level
+    ...
+    @property
+    def _array_length_level(self):
+        return array_length_level
+```
+
+Any dialect-specific version of comparisons will inherit from this (where it learns about the dialect-specific comparison levels), and the comparison itself; in our case, in the same file:
+
+```python
+...
+class array_length_at_thresholds(
+    SparkComparisonProperties, ArrayLengthAtThresholds
+):
+    pass
+```
+
+This is now ready to import and be used, just as any other pre-existing comparisons.
+
+### Subclassing existing library comparisons
+
+As in our `hamming_level` example above, you can similarly subclass existing library `Comparison`s. The main difficulty here is that the subclassed comparison levels will have different function arguments, so we need to check within the constructor if we are in the generic version (`distance_function_at_thresholds`) or a specific version (`hamming_level`). See the [source code for `DistanceFunctionAtThreholdsComparisonBase`](https://github.com/moj-analytical-services/splink/blob/master/splink/comparison_library.py) for an example.
```

Comment:
Here I have not included much detail as I think the current way of doing things is a bit messy, and there is probably a better way, and feel like this is probably unlikely to be a particularly common use-case for the time being

Reply:
That's fine and I think you've elaborated enough above for users to gain enough into how this process is done.